embed-code-go/embedding/parsing/constants.go

```diff
@@ -0,0 +1,25 @@
+package parsing
+
+// Maps state names to the list of possible next states.
+//
+// The order of the next states is important.
```

Comment:
Lets not say they are just `important` describe why it is so. You should say here that the `next state` is checked one by one so far if the first state matches the criteria it means that the other won't be checked at all. 

Imagine in two years you should add some new step here you don't remember this code, and what we've discussed here. As a result doc here should explain to you where to put the new step and why.